department-of-veterans-affairs/developer-portal
# Separate API category pages and remove hero on documentation pages

## Overview

PR for ticket 1157: https://github.com/department-of-veterans-affairs/vets-contrib/issues/1157

This branch contains changes to separate API category pages. Previously, the pages for Benefits, Facilities, Health, and Veteran Verification were all rendered on the same page and scrolled to match the correct route. Now, each API category gets its own page. There is a new documentation overview page at `/explore` used by the Overview link at the top of side nav. This page and each API category page now have their links styled in cards and laid out horizontally across the top of the page. The hero on the documentation pages has also been removed.

## Testing

* Verify that the API category pages are rendered alone on their respective routes and that the documentation page exists. That means you should check the following routes:
    * `/explore`
    * `/explore/benefits`
    * `/explore/health`
    * `/explore/facilities`
    * `/explore/verification`
* Content should be left-aligned and appropriately spaced on all pages.
* Verify the layout on both mobile and desktop, as well as responsive sizing at intermediate sizes.
* Headings on documentation pages should be in [Bitter font](https://fonts.google.com/specimen/Bitter).
* The documentation for specific API pages (ie Benefits Intake, Appeals Status, Claims Status for Benefits) should remain the same as before, with the exception of left-alignment and header font.
* The cards on the documentation overview page and API category pages should wrap neatly as the screen shrinks.

### Desktop

* The documentation overview should appear like this: 

<img width="1413" alt="documentationoverviewdesktop" src="https://user-images.githubusercontent.com/3241493/53746088-76162c00-3e6e-11e9-93c0-e44f56235678.png">

* For each API category page, the page should have a header, subheader, and then the cards for the specific APIs. Below that is the content previously included in the API category page. 

<img width="1413" alt="benefitsapipagedesktop" src="https://user-images.githubusercontent.com/3241493/53768986-7d5a2b80-3ea8-11e9-81ad-d263855674f6.png">

* When the screen width is wider than the page content `max-width` of 1200px, there should be 3 tiles in one row.

### Mobile

* The documentation overview should appear like this: 

<img width="340" alt="documentationoverviewmobile" src="https://user-images.githubusercontent.com/3241493/53769097-de81ff00-3ea8-11e9-92fd-59d8ab91c24a.png">

* For each API category page, the page should have a header, subheader, and then the cards for the specific APIs. Because it is mobile, the cards will be stacked vertically and take up the whole width. Below that is the content previously included in the API category page. 

<img width="342" alt="benefitsapipagemobile" src="https://user-images.githubusercontent.com/3241493/53769075-cf9b4c80-3ea8-11e9-97a0-ec59ab5cda59.png">

* At 767px and below, the cards on any given page should take the full width minus padding. That is, any time the cards wrap so that there is only one card per row, they should take up the whole space.

## Notes

* Any suggestions for unit or e2e testing welcome.
* The saving of the old version of `ApiPage` as `ApiPageWithScroll` is intentional; see the AC on the ticket.
